### PR TITLE
[FlexibleHeader] Always update opacity for views that hide when shifted.

### PR DESCRIPTION
**Pull request by jverkoey**
Before this change, views that hide when shifted would only change their visibility if shifting was enabled. This resulted in a bug where if shifting was disabled then views would not become visible again once the header shifted back on screen.

After this change, views that hide when shifted will always change their visibility when the header's frame changes.

Closes https://github.com/material-components/material-components-ios/issues/3559

### Before
before

### After

after

### CODE REVIEW

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m [hunk 1]
```diff
@@ -1052,6 +1042,15 @@ - (void)fhv_commitAccumulatorToFrame {
   [self fhv_accumulatorDidChange];
   [self fhv_recalculatePhase];
 
+  // Update opacity of _viewsToHideWhenShifted
+  CGFloat contentHeight = self.computedMinimumHeight - MDCDeviceTopSafeAreaInset();
+  CGFloat hideThreshold = kContentHidingThreshold;
+  CGFloat boundedAccumulator = MIN([self fhv_accumulatorMax], _shiftAccumulator);
+  CGFloat alpha = MAX(contentHeight - boundedAccumulator / hideThreshold, 0) / contentHeight;
+  for (UIView *view in _viewsToHideWhenShifted) {
```
Comment (mohammadcazig) [inline components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Could you add more comments here to clarify what this code is trying to do.

**Review by mohammadcazig — DISMISSED** [review]
Documentation

**Comment by jverkoey** [comment]
Filed https://github.com/material-components/material-components-ios/issues/4183 to track other bug discovered in this PR.

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m (hunk 1 above)
Comment (jverkoey) [reply components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Done.

**Review by ianegordon — APPROVED** [review]

**Pull request merged**